theano/tensor/tests/test_opt.py

```diff
@@ -3812,6 +3812,56 @@ def test(self):
         assert isinstance(topo[0].op, T.Elemwise)
 
 
+class T_func_inverse(unittest.TestCase):
+
+    def setUp(self):
+        mode = theano.compile.get_default_mode()
+        self.mode = mode.including('local_func_inv')
+
+    def test(self):
+        """
+        test that consecutive ops that are functional inverses are removed
+        """
+
+        x = T.fmatrix()
+        o = T.deg2rad(T.rad2deg(x))
+        f = theano.function([x], o, mode=self.mode)
+        dx = numpy.random.rand(5, 4).astype("float32")
+        delta = f(dx) - dx
+        topo = f.maker.fgraph.toposort()
+
+        self.assertEqual(len(topo), 1)
+        self.assertTrue(numpy.all(delta == 0))
+        self.assertTrue(isinstance(topo[0].op, DeepCopyOp),
+                        "Inverse functions not removed!")
+
+        # Test that the other ordering of functions works
+        x = T.fmatrix()
+        o = T.rad2deg(T.deg2rad(x))
+        f = theano.function([x], o, mode=self.mode)
+        dx = numpy.random.rand(5, 4).astype("float32")*180
+        delta = f(dx) - dx
+        topo = f.maker.fgraph.toposort()
+
+        self.assertEqual(len(topo), 1)
+        self.assertTrue(numpy.all(delta == 0))
+        self.assertTrue(isinstance(topo[0].op, DeepCopyOp),
+                        "Inverse functions not removed!")
+
+        # Test that non-inverse functions are ran normally
+        x = T.fmatrix()
+        o = T.rad2deg(T.rad2deg(x))
+        f = theano.function([x], o, mode=self.mode)
+        dx = numpy.random.rand(5, 4).astype("float32")+0.01
+        delta = f(dx) - dx
+        topo = f.maker.fgraph.toposort()
+
+        self.assertEqual(len(topo), 1)
```

Comment:
The travis log are here: https://s3.amazonaws.com/archive.travis-ci.org/jobs/58345808/log.txt

The failing test:
```
======================================================================
FAIL: test that consecutive ops that are functional inverses are removed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/pyenv/lib/python2.6/site-packages/theano/tensor/tests/test_opt.py", line 3859, in test
    self.assertEqual(len(topo), 1)
AssertionError: 2 != 1
----------------------------------------------------------------------
```

In FAST_COMPILE, the topo have a lenght of 2, so you could just relax this assert to test for the len of topo to be 1 or 2.

Thanks for the PR.